src/snowflake/snowpark/dataframe_na_functions.py

```diff
@@ -206,7 +200,7 @@ def drop(
                     df_col_type_dict[normalized_col_name], (FloatType, DoubleType)
                 ):
                     # iff(col = 'NaN' or col is null, 0, 1)
-                    is_na = iff((col == "NaN") | col.is_null(), 0, 1)
+                    is_na = iff((col == math.nan) | col.is_null(), 0, 1)
```

Comment:
it feels like this change will affect the live connection or it won't?

Reply:
It won't, `math.nan` gets converted to `NaN::float` in the generated query.